addons/purchase/tests/test_purchase.py

```diff
@@ -559,3 +559,39 @@ def test_purchase_order_multi_company(self):
         po = po_form.save()
 
         self.assertEqual(po.order_line[0].price_unit, 3.0)
+
+    def test_purchase_suggest_qty(self):
+        """
+        Checks the suggested qty of POL is correctly set based on valid supplier-info
+        leading to correctly compute the price unit, product_qty and product_desc
+        """
+        product = self.env['product.product'].create({
+            'name': 'SuperProduct',
+            'type': 'consu',
+            'standard_price': 100,
+        })
+        self.env['product.supplierinfo'].create({
+            'partner_id': self.partner_a.id,
+            'product_id': product.id,
+            'min_qty': 1,
+            'price': 50,
+            'date_start': fields.Date.today() - timedelta(days=5),
+            'date_end': fields.Date.today() - timedelta(days=3),
+            'product_code': 'product_code_1',
+        })
+        self.env['product.supplierinfo'].create({
+            'partner_id': self.partner_a.id,
+            'product_id': product.id,
+            'min_qty': 10,
+            'price': 100,
+            'date_start': fields.Date.today() - timedelta(days=5),
+            'date_end': fields.Date.today() + timedelta(days=3),
+            'product_code': 'HHH',
+        })
+        po_form = Form(self.env['purchase.order'])
+        po_form.partner_id = self.partner_a
+        with po_form.order_line.new() as po_line:
+            po_line.product_id = product
+        po = po_form.save()
+        self.assertEqual(po.order_line.product_qty, 10.0)
+        self.assertEqual(po.order_line.name, '[HHH] SuperProduct')
```

Comment:
I don't agree with the targeted behavior
I don't really see why we would define a quantity and therefore match all supplier info